core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -641,7 +641,8 @@ private void doApplyState(Address sender, int segmentId, Collection<InternalCach
                ctx = icf.createSingleKeyNonTxInvocationContext();
             }
 
-            PutKeyValueCommand put = commandsFactory.buildPutKeyValueCommand(e.getKey(), e.getValue(), segmentId,
+            // CallInterceptor knows PUT_FOR_STATE_TRANSFER values are actually InternalCacheEntries
+            PutKeyValueCommand put = commandsFactory.buildPutKeyValueCommand(e.getKey(), e, segmentId,
```

Comment:
The only thing I wonder is, we could be duplicating the metadata we send across right? Maybe we should pass null for metadata?

Reply:
These put commands are never sent over the network.